code/modules/mob/living/carbon/examine.dm

```diff
@@ -88,6 +88,7 @@
 	return ""
 
 /mob/living/carbon/examine(mob/user)
+	var/hallucinating = has_status_effect(STATUS_EFFECT_HALLUCINATION)
```

Comment:
you need to make this a new hallucination, and not check if theyre hallucinating at all, like https://github.com/ParadiseSS13/Paradise/pull/27616. They used traits.